components/style/selector_matching.rs

```diff
@@ -252,25 +253,62 @@ impl Stylist {
 
     /// Computes the style for a given "precomputed" pseudo-element, taking the
     /// universal rules and applying them.
+    ///
+    /// If `inherit_all` is true, then all properties are inherited from the parent; otherwise,
+    /// non-inherited properties are reset to their initial values. The flow constructor uses this
+    /// flag when constructing anonymous flows.
     pub fn precomputed_values_for_pseudo(&self,
                                          pseudo: &PseudoElement,
-                                         parent: Option<&Arc<ComputedValues>>)
+                                         parent: Option<&Arc<ComputedValues>>,
+                                         inherit_all: bool)
                                          -> Option<Arc<ComputedValues>> {
         debug_assert!(TheSelectorImpl::pseudo_element_cascade_type(pseudo).is_precomputed());
         if let Some(declarations) = self.precomputed_pseudo_element_decls.get(pseudo) {
+            let mut flags = CascadeFlags::empty();
+            if inherit_all {
+                flags.insert(INHERIT_ALL)
+            }
+
             let (computed, _) =
                 properties::cascade(self.device.au_viewport_size(),
-                                    &declarations, false,
+                                    &declarations,
                                     parent.map(|p| &**p),
                                     None,
                                     None,
-                                    Box::new(StdoutErrorReporter));
+                                    Box::new(StdoutErrorReporter),
+                                    flags);
             Some(Arc::new(computed))
         } else {
             parent.map(|p| p.clone())
         }
     }
 
+    /// Returns the style for an anonymous box of the given type.
+    pub fn style_for_anonymous_box(&self,
```

Comment:
You should mark this function as `#[cfg(feature = "servo")]` so this compiles on Gecko.